unit-tests-jest/numbers.js

```diff
@@ -0,0 +1,54 @@
+/**
+ * Returns a new array that contains only the even numbers in the given array.
+ */
+export function evenNumbers(numbers) {
+  return numbers.filter((n) => n % 2 === 0);
+}
+
+/**
+ * Returns a number formatted in dollars and cents.
+ * It works for both integer and fractional numbers.
+ * That is, `toDollars(12)` returns `$12.00` and
+ * `toDollars(11.341)` returns `$11.34`.
+ */
+export function toDollars(amount) {
+  if (
+    amount === undefined ||
+    amount === null ||
+    isNaN(amount) ||
+    amount === 0 ||
+    !amount
+  )
+    return 'Invalid value try again';
+  if (Number.isInteger(amount)) return `$${amount}.00`;
+  return `$${Math.round(amount * 100) / 100}`;
+}
+
+/**
+ * Returns a new array of numbers where every entry has been divided by the
+ * given divisor. Does not modify the original array.
+ * @param {Array} numbers The array of numbers to be divided by the divisor.
+ * @param {number} divisor The number to divide into each element of `numbers`.
+ * @returns a new array.
+ */
+export function divideBy(numbers, divisor) {
+  for (let i = 1; i < numbers.length; i++) {
+    numbers[i] = numbers[i] / divisor;
+  }
+  return numbers;
```

Comment:
The function description says it should not modify the input, but this function does modify it. Using `.map` you can easily return a new array.